sklearn/tree/_splitter.pxd

```diff
@@ -28,7 +28,8 @@ cdef struct SplitRecord:
     float64_t lower_bound     # Lower bound on value of both children for monotonicity
     float64_t upper_bound     # Upper bound on value of both children for monotonicity
     unsigned char missing_go_to_left  # Controls if missing values go to the left node.
-    intp_t n_missing       # Number of missing values for the feature being split on
+    intp_t n_missing            # Number of missing values for the feature being split on
+    intp_t n_constant_features  # Number of constant features in the split from parent
```

Comment:
Looking at the semantic of the split, I think `n_constant_features` is not natural to be included here. A split is the info for a specific feature while `n_constant_features` is an overall for all feature. For instance, a variable that would be `is_constant` would follow the expected semantic of the split (but would be useless for what we try to achieve).

So this `n_constant_features` should be store at a higher level data structure where we have the info of all features, so it might only be the tree in this case.